src/Raven.Studio/typescript/viewmodels/database/query/spatialQueryMap.ts

```diff
@@ -0,0 +1,142 @@
+import viewModelBase = require("viewmodels/viewModelBase");
+import spatialMarkersLayerModel = require("models/database/query/spatialMarkersLayerModel");
+import document = require("models/database/documents/document");
+import documentMetadata = require("models/database/documents/documentMetadata");
+import {Control, LatLngExpression, Layer, MarkerClusterGroup} from "leaflet";
+import genUtils = require("common/generalUtils");
+import spatialCircleModel = require("models/database/query/spatialCircleModel");
+import spatialPolygonModel = require("models/database/query/spatialPolygonModel");
+
+class spatialQueryMap extends viewModelBase {
+    
+    markersLayers = ko.observableArray<spatialMarkersLayerModel>([]);
+    circlesLayer = ko.observableArray<spatialCircleModel>([]);
+    polygonsLayer = ko.observableArray<spatialPolygonModel>([]);
+
+    constructor(markers: Array<spatialMarkersLayerModel>, circles: spatialCircleModel[], polygons: spatialPolygonModel[]) {
+        super();
+        this.markersLayers(markers);
+        this.circlesLayer(circles);
+        this.polygonsLayer(polygons);
+    }
+
+    compositionComplete() {
+        super.compositionComplete();
+        this.createMap();
+    }
+    
+    private createMap() {
+
+        const osmMap = this.getStreetMapTileLayer();
+        const otmMap = this.getTopographyMapTileLayer();
+        const baseLayers =  {
+            "Streets Map": osmMap,
+            "Topography Map": otmMap
+        };
+
+        const generatePopupContent = (doc: document) => {
+            const docDto = doc.toDto(true);
+            const metaDto = docDto["@metadata"];
+            documentMetadata.filterMetadata(metaDto);
+
+            let text = JSON.stringify(docDto, null, 4);
+            text = Prism.highlight(text, (Prism.languages as any)["javascript"]);
+
+            const textHtml = `<div>
+                                    <h4>Document: ${genUtils.escapeHtml(doc.getId())}</h4>
+                                    <hr>
+                                    <pre>${text}</pre>
+                              </div>`;
+            return textHtml;
+        }
+        
+        const dataLayers: Control.LayersObject = {};
+        const markersGroups: MarkerClusterGroup[] = [];
+
+        this.markersLayers().forEach((markersLayer) => {
+            const markers: Layer[] = [];
+            const markersGroup = L.markerClusterGroup();
+
+            markersLayer.geoPoints().forEach((point) => {
+                const pointMarker = L.marker([point.Latitude, point.Longitude], { title: point.PopupContent.getId() })
+                                     .bindPopup(generatePopupContent(point.PopupContent), { 'className' : 'custom-popup', 'maxWidth': 600, 'maxHeight': 400 } );
```

Comment:
you generate popup content from popup content? I think we should have better name for PopupContent

Reply:
`generatePopupContent` => `generatePopupHtml`